mongo/gridfs_bucket.go

```diff
@@ -190,26 +190,23 @@ func (b *GridFSBucket) DownloadToStream(ctx context.Context, fileID interface{},
 func (b *GridFSBucket) OpenDownloadStreamByName(
 	ctx context.Context,
 	filename string,
-	opts ...*options.NameOptions,
+	opts ...Options[options.GridFSNameArgs],
 ) (*GridFSDownloadStream, error) {
-	var numSkip int32 = -1
-	var sortOrder int32 = 1
+	args, err := newArgsFromOptions[options.GridFSNameArgs](opts...)
+	if err != nil {
+		return nil, fmt.Errorf("failed to construct arguments from options: %w", err)
+	}
 
-	nameOpts := options.GridFSName()
-	nameOpts.Revision = &options.DefaultRevision
+	var numSkip int32 = -1
```

Comment:
Shall we initialize `numSkip` with `&options.DefaultRevision`?

Reply:
That makes sense. Looking at this method a bit closer, it's unclear why we set the Revision on the arguments. Suggest we also remove that code.